homeassistant/components/blue_current/sensor.py

```diff
@@ -0,0 +1,342 @@
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        name="Grid Current Phase 2",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+    ),
+    SensorEntityDescription(
+        key="grid_actual_p3",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        name="Grid Current Phase 3",
+        entity_registry_enabled_default=False,
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+    ),
+    SensorEntityDescription(
+        key="grid_avg_current",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        name="Average Grid Current",
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+    ),
+    SensorEntityDescription(
+        key="grid_max_current",
+        native_unit_of_measurement=UnitOfElectricCurrent.AMPERE,
+        device_class=SensorDeviceClass.CURRENT,
+        name="Max Grid Current",
+        state_class=SensorStateClass.MEASUREMENT,
+        has_entity_name=True,
+    ),
+)
+
+PARALLEL_UPDATES = 1
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Blue Current sensors."""
+    connector: Connector = hass.data[DOMAIN][entry.entry_id]
+    sensor_list: list[SensorEntity] = []
+    for evse_id in connector.charge_points:
+        for sensor in SENSORS:
+            sensor_list.append(ChargePointSensor(connector, sensor, evse_id))
+
+    for grid_sensor in GRID_SENSORS:
+        sensor_list.append(GridSensor(connector, grid_sensor))
+
+    async_add_entities(sensor_list)
+
+
+class ChargePointSensor(BlueCurrentEntity, SensorEntity):
+    """Define a charge point sensor."""
+
+    _attr_should_poll = False
+
+    def __init__(
+        self,
+        connector: Connector,
+        sensor: SensorEntityDescription,
+        evse_id: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(connector, evse_id)
+
+        self.key = sensor.key
+        self.entity_description = sensor
+        self._attr_unique_id = f"{sensor.key}_{evse_id}"
+
+    @callback
+    def update_from_latest_data(self) -> None:
+        """Update the sensor from the latest data."""
+
+        new_value = self.connector.charge_points[self.evse_id].get(self.key)
+
+        if new_value is not None:
+            if self.key in TIMESTAMP_KEYS and not (
+                self._attr_native_value is None or self._attr_native_value < new_value
+            ):
+                return
+            self._attr_available = True
+            self._attr_native_value = new_value
+
+        elif self.key not in TIMESTAMP_KEYS:
+            self._attr_available = False
+
+
+class GridSensor(SensorEntity):
```

Comment:
I would recommend to make the BlueCurrentEntity more generic so it can be a pure base class for all entities. You can define `_attr_should_poll = False` on that level for example.

Reply:
Added `ChargepointEntity` that inherits `BlueCurrentEntity` and moved the device info stuff there. This way the grid sensor can inhert `BlueCurrentEntity` and the chargepoint sensors the `ChargepointEntity`. Also moved `_attr_should_poll` and `_attr_has_entity_name` to `BlueCurrentEntity`.